### PR TITLE
Split up filters in smaller components

### PR DESCRIPTION
**Pull request by rcantin-w**
## Who is this for?
Me, mostly 😅 

## What is it doing for them?
Splits up `SearchFiltersDesktop` into smaller components to help with readability / understanding where what is used. Filters currently being used in
- Desktop pill format
- Desktop modal
- SearchFiltersMobile
- No JS
- Old search AND new search with slight differences

It was getting hard to wrap my head around so this is a step in an eventual cleanup post deleting "old" search pages. With `common/views/components/SearchFilters/SearchFiltersDesktop.tsx` becoming `common/views/components/SearchFilters/SearchFilters.Desktop.tsx`, which doesn't seem clear here.

No code was changed, just splitting up, changing names and updating imports.

### CODE REVIEW

File: catalogue/webapp/pages/search/images.tsx [hunk 1]
```diff
@@ -10,7 +10,7 @@ import Space from '@weco/common/views/components/styled/Space';
 import SearchNoResults from '@weco/catalogue/components/SearchNoResults/SearchNoResults';
 import SearchContext from '@weco/common/views/components/SearchContext/SearchContext';
 import Pagination from '@weco/common/views/components/Pagination/Pagination';
-import SearchFilters from '@weco/common/views/components/SearchFilters/SearchFilters';
+import SearchFilters from '@weco/common/views/components/SearchFilters';
```
Comment (alexwlchan) [inline catalogue/webapp/pages/search/images.tsx:13]: Big fan of these new import conventions. 👍

File: common/views/components/SearchFilters/SearchFilters.Desktop.CheckboxFilter.tsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+      buttonType="inline"
+      id={f.id}
+      hasNoOptions={f.options.length === 0}
+    >
+      <PlainList className={font('intr', 5)}>
+        {f.options.map(({ id, label, value, count, selected }, i) => {
+          return (
+            // TODO remove index from key once we resolve the doubled IDs issue
+            // (https://github.com/wellcomecollection/wellcomecollection.org/issues/9109)
+            // as we now sometimes get "Warning: Encountered two children with the same key" console errors
```
Comment (alexwlchan) [inline common/views/components/SearchFilters/SearchFilters.Desktop.CheckboxFilter.tsx:38]: Good comment. 👍

File: common/views/components/ModalMoreFilters/ModalMoreFilters.tsx [hunk 3]
```diff
@@ -16,7 +16,7 @@ import { AppContext } from '@weco/common/views/components/AppContext/AppContext'
 import CheckboxRadio from '@weco/common/views/components/CheckboxRadio/CheckboxRadio';
 import PlainList from '@weco/common/views/components/styled/PlainList';
 import { LinkProps } from '@weco/common/model/link-props';
-import DateRangeFilter from '@weco/common/views/components/SearchFilters/SearchFilters.DateRange';
+import DateRangeFilter from '@weco/common/views/components/SearchFilters/SearchFilters.DateRangeFilter';
```
Comment (alexwlchan) [inline common/views/components/ModalMoreFilters/ModalMoreFilters.tsx:19]: I wonder: should we export this (and all other components used externally) from `@weco/common/views/components/SearchFilters/index.tsx`, so external callers don't have to worry about the exact structure of this component?

**Review by alexwlchan — APPROVED** [review]
I haven't read the code in detail but I trust it's the same; just a suggestion about going even further on our improved approach to imports.

File: common/views/components/ModalMoreFilters/ModalMoreFilters.tsx (hunk 3 above)
Comment (rcantin-w) [reply common/views/components/ModalMoreFilters/ModalMoreFilters.tsx:19]: That's a nice idea, I'll look into that change

**Pull request merged**